.github/workflows/pull-request-main.yml

```diff
@@ -39,6 +39,43 @@ jobs:
           gc-basic-auth: ${{ secrets.GRAFANA_CLOUD_BASIC_AUTH }}
           gc-host: ${{ secrets.GRAFANA_CLOUD_HOST }}
 
+  cd-signed-commits:
```

Comment:
Should this depend on `ci-signed-commits` so we don't "deploy" failing code?

Reply:
that's an option, it would serialize the two jobs though and make the feedback loop slower. The building of the action itself can surface errors that need to be fixed. I think it's better that instead we add both of these checks as required checks in order to merge to trunk

Reply:
> that's an option, it would serialize the two jobs though and make the feedback loop slower.

I don't think it would slow the feedback loop enough to cause many issues. My main problem with it is the automation of commits for code that hasn't passed CI.

Why add that to the commit history?

Reply:
IMO we're at a edge case b/c usually you never commit artifacts to a repository in the first place. I think there's value in trying to build anyway, we often run docker builds as a CI check. I see your point about commit history pollution, but if it becomes a large problem then we could squash the PR.

I think its because i see it as a CI check of its own which is why IDM leaving it in the way it is. I'd have a problem with it if it was auto deploying too, but we're not doing that, nor are we affecting other people outside of this PR scope.

I also think we should always have CI as fast as possible, shorter loops have an outsized impacts on dev workflow (1 min feedback vs 2 min is actually a huge difference)

Reply:
> I think its because i see it as a CI check of its own which is why IDM leaving it in the way it is. 

I'm not against building in CI, I'm against automatically committing compiled code to a PR. In fact, we're already building as part of a pull request CI.

- https://github.com/smartcontractkit/.github/blob/edf5b41acd7ae24807396c43c967800415f14cd1/.github/workflows/pull-request-main.yml#L65
- https://github.com/smartcontractkit/.github/blob/edf5b41acd7ae24807396c43c967800415f14cd1/actions/cicd-build-publish-artifacts-ts/action.yml#L109
- https://github.com/smartcontractkit/.github/blob/edf5b41acd7ae24807396c43c967800415f14cd1/package.json#L10C1-L10C58

Reply:
good point, lets serialize it then

Reply:
Resolving this b/c we already do a build inside the tests themselves, oops :see_no_evil:

Reply:
Discussed offline. 

Will follow-up with an updated workflow for building + committing outputs for all typescript actions with more detailed messages.

Reply:
See 70e65bbc3f1926119ed93a2f4962cf2d7198cc61 and #78.